client/src/NationalStats.tsx

```diff
@@ -4,22 +4,21 @@ import CanadaNationalStats from './Models/CanadaNationalStats';
 export default class NationalStats extends Component<{longitude: any, latitude: any}, {canadaNationalStats: CanadaNationalStats}> {
     async componentDidMount(): Promise<void> {
     }
-    async componentWillReceiveProps(nextProps: {latitude: any, longitude: any}): Promise<void> {
-        if (nextProps.latitude !== 0 && nextProps.longitude !== 0) {
-            this.setState({canadaNationalStats: await this.getLocationFromPosition(nextProps.latitude, nextProps.longitude)});
+    async componentDidUpdate(prevProps: {longitude: any, latitude: any}): Promise<void> {
+        if (prevProps.longitude !== this.props.longitude && prevProps.latitude !== this.props.latitude) {
+            this.setState({canadaNationalStats: await this.getLocationFromPosition(this.props.longitude, this.props.latitude)});
         }
     }
     componentWillUnmount() {
-        // fix Warning: Can't perform a React state update on an unmounted component
```

Comment:
I kind of want to keep this comment, otherwise I think this seemingly useless method might get deleted.